.github/CODESTYLE.md

```diff
@@ -0,0 +1,305 @@
+  * Derived type arguments only need documented in bulk, their individual components only need defined where the derived type is defined.
+  * Variable descriptions will not be redundant with the variable name
+  * Descriptions will describe the physical quantity
+  * Descriptions will include units.
+* Variable names shall contain a noun.
+* Exception: when widely accepted equations have simple variable names. (e.g. `F=m*a`)
+* Loop counters should typically have names.
+* Loop counters should NOT be single letters. Simply doubling the letter makes it easier to search for the use (and possibly replace).
+* Conventions for special kinds of variables:
+  * Accumulation variables: “acc” prefix, e.g.
+  * MPI global variables
+  * prognostic vs diagnostic variables
+  * logical variables must be named to reflect the state they are used to convey, most with the verb to be, e.g. :
+    * `lib_is_initialized` vs `lib_init`
+    * `obj_has_parent` vs `obj_parent`
+
+## Function/Subroutine Names
+* Function names shall contain at least one verb, e.g. `get_name`, `parse_control_string`.
+* Function names shall convey their intended use to other developers who did not author the code.
+* Follow the `lowercase_underscore_separated` convention.
+
+## Constant names
+* Name all constants: Numbers should not be used in the code except in variable definitions.
+* All constants are defined as parameters.
+* Exceptions: should be rare and clearly typed (float, double, etc).
+* Constants follow the `UPPERCASE_UNDERSCORE SEPARATED` convention for naming.
+* Constant names shall describe what the contained value represents within context, e.g. `NO_DATA_VALUE`, `PLANCK_LENGTH`, `HIGH_TEMPERATURE_THRESHOLD`, etc.
+
+# Documenting in the source
+* Code documentation should be a primary source of overall documentation. By following simple practices, such documentation can be extracted to live separately if needed. But keeping it close as possible to the source maintains consistency and improves communication among developers: all you need is the source to understand the code (not other random documents)
+* Write code that documents itself. Try to make code as clear as possible (“self-documenting”) to avoid use of comments and redundancy between the two which needs synchronized or is confusingly out of sync. Writing self-documenting code includes using variable names with obvious meaning and documenting ambiguities in variable names when they exist. Minimize commenting “what” (repeating the code with comments) but do it when it is necessary for interpretation.
+* Document why (not what). Algorithmic choices are often the hardest thing to perceive, not function calls on variables.
+* Indent comments to the indentation level of the code which is being commented.
+* Comments shall be written in English with good spelling, punctuation, and grammar.
+* Documentation shall be placed in the code and FORD will be used to generate documentation.
```

Comment:
Would a link to FORD standards help here?

Reply:
added link